src/commands/CommandRouter.ts

```diff
@@ -0,0 +1,150 @@
+        });
+
+        this._eventRouter.addEventListener(Events.MessageReactionAdd, async (reaction: any, user: any) => {
+            const { message, emoji } = reaction;
+            const alreadyReacted = (reaction.count && reaction.count > 1) == true;
+        
+            //dangerous - allow seabot to react to a bot's commands for creating server events
+            if (
+                !alreadyReacted &&
+                message.author?.id === UserIDs.APOLLO &&
+                emoji.name === CreateServerEvent &&
+                isModReaction(reaction, user) &&
+                !Environment.DEBUG
+            ) {
+                const event = CreateServerEvent.execute
+                    ? ((await CreateServerEvent.execute(message as Message)) as GuildScheduledEvent)
+                    : null;
+                try {
+                    if (event) {
+                        //confirm event creation
+                        await message.react("✅");
+                        //attempt to message user who created it with link
+                        await user.send(`Event created: ${event.url}`);
+                    }
+                } catch (e: any) {
+                    console.dir(e);
+                } finally {
+                    //stop processing  reactions after this
+                    return;
+                }
+            }
+        
+            // this prevents the same reaction command from firing multiple times
+            if (message.author?.bot || !emoji.id || alreadyReacted) {
+                return;
+            }
+
+            try {
+                if (commandMap.has(emoji.name)) {
+                    const command = commandMap.get(emoji.id) as ReactionCommand;
+                    if (command.execute) {
+                        command.execute(message);
+                    }
+
+                    if (command.removeReaction) {
+                        await reaction.remove();
+                    }
+                }
+            } catch (e: any) {
+                console.dir(e);
+                message.react("💩");
+            }
+        });
+    }
+
+    private async listenForContentCommands(contentCommands: ContentCommand[]) {
+        this._eventRouter.addEventListener(Events.MessageCreate, async (message: Message) => {
+            if (message.author.bot) return;
+            if (message.partial) message = await message.fetch();
+
+            for (const command of contentCommands) {
+                try {
+                    if ((command.trigger instanceof RegExp)) {
+                        if (command.trigger.test(message.content)) {
+                            command.handler(message);
+                        }
+                    } else if (message.content === command.trigger) {
+                        command.handler(message);
+                    }
+                } catch (error) {
+                    console.error(`Error while handling command ${command.trigger}`);
```

Comment:
Would it be useful to output the message as well?

Reply:
Agree. This is also something we can further address in a follow-up PR re: better error handing/logging